modal/shared/logging.py

```diff
@@ -31,6 +33,17 @@ def add_observability(image: Image):
     )
 
 
+@contextmanager
+def timer(action: str, tags: dict[str, str | int] = None) -> None:
+    """A simple timer context manager with structured logging for its output."""
+    start = time.perf_counter()
+    yield
+    elapsed = time.perf_counter() - start
+
+    extra = tags or {} | {"duration": elapsed}
```

Comment:
```suggestion
    extra = (tags or {}) | {"duration": elapsed}
```
to fix the precedence

Reply:
oo nice catch :pray: